src/Data/Bifunctor/Sum.hs

```diff
@@ -17,15 +17,18 @@ import Data.Bifoldable
 import Data.Bitraversable
 import Data.Data
 import Data.Functor.Classes
-import GHC.Generics
+import GHC.Generics (Generic, Generic1)
+import Text.Read ((+++))
 
 data Sum p q a b
   = L2 (p a b)
   | R2 (q a b)
   deriving (Eq, Ord, Show, Read, Data, Generic, Generic1, Functor, Foldable, Traversable)
 
-instance (Eq2 f, Eq2 g, Eq a) => Eq1 (Sum f g a) where
-  liftEq = liftEq2 (==)
+instance (Eq1 (p a), Eq1 (q a)) => Eq1 (Sum p q a) where
+  liftEq eq (L2 x) (L2 y) = liftEq eq x y
+  liftEq eq (R2 x) (R2 y) = liftEq eq x y
+  liftEq _ _ _ = False
```

Comment:
An extremely minor point, but the `liftEq2` implementation has explicit cases for `L2 + R2` and `R2 + L2`, while the `liftEq` implementation doesn't. I'd prefer for them to be uniform, even if only for the sake of stylistic consistency.